### DISCUSSION TITLE
How do I access the WWNs via the transport field in a Fibre Channel path?

### DISCUSSION THREAD

**Discussion by ericwest-boston** [vmware/govmomi#3317] [Community Help] [answered]
`	for _, lun := range hss.StorageDeviceInfo.MultipathInfo.Lun {
		for _, path := range lun.Path {
			fmt.Printf("FC Lun path xport:     %v\n", path.Transport)
}`
 
 Will print out:
 FC Lun path xport:     &{{{}} 2769714599750104683 2341872635149907563}
 
 However, having tried more than a few approaches, I am unable to access the two WWN values 
 
 Interesting if using the VS Code debugger, in the debug window:
 `path.Transport
github.com/vmware/govmomi/vim25/types.BaseHostTargetTransport(*github.com/vmware/govmomi/vim25/types.HostFibreChannelTargetTransport) *{HostTargetTransport: github.com/vmware/govmomi/vim25/types.HostTargetTransport {DynamicData: github.com/vmware/govmomi/v...
data:
*github.com/vmware/govmomi/vim25/types.HostFibreChannelTargetTransport {HostTargetTransport: github.com/vmware/govmomi/vim25/types.HostTargetTransport {DynamicData: github.com/vmware/govmomi/vim25/types.DynamicData {}}, PortWorldWideName: 27697145997501046...
github.com/vmware/govmomi/vim25/types.HostFibreChannelTargetTransport {HostTargetTransport: github.com/vmware/govmomi/vim25/types.HostTargetTransport {DynamicData: github.com/vmware/govmomi/vim25/types.DynamicData {}}, PortWorldWideName: 276971459975010468...
  HostTargetTransport:
github.com/vmware/govmomi/vim25/types.HostTargetTransport {DynamicData:github.com/vmware/govmomi/vim25/types.DynamicData {}}
    PortWorldWideName: 2769714599750104683
    NodeWorldWideName: 2341872635149907563`

...and I can access the WWNs directly in that console
`path.Transport.NodeWorldWideName
2341872635149907563`

So how do I do this programmatically?   ex `fmt.Println(path.Transport.NodeWorldWideName)`
Note too: VS Code claims NodeWorldWideName is not a field for Transport

Thanks!

**Comment by dougm** [comment, marked as answer]
`path.Transport` here is base type [HostTargetTransport](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.host.TargetTransport.html), which has 8 subtypes. You can use Go's type assertions to access specific fields:

```go
		for _, path := range lun.Path {
			switch xport := path.Transport.(type) {
			case *types.HostFibreChannelTargetTransport:
				fmt.Printf("wwn=%d\n", xport.PortWorldWideName)
			default:
				fmt.Printf("type=%T\n", xport)
			}
		}
```

**Reply by ericwest02** [reply]
Doh!   of course. 

Intuitively obvious :)

Thanks!   -- I might still have some hair left :p
